README.md

```diff
@@ -17,6 +17,78 @@ the config file `ledger2beancount.yml` and if that is not found for
 file for the variables.
 
 
+FUNCTIONALITY
+-------------
+
+### Narration
+
+The ledger payee information, which is generally used as free-form text
+to describe the transaction, is stored in beancount's narration field
+and properly quoted.
+
+### Payees
+
+Ledger has limited support for payees.  A `payee` metadata key can be set
+but this also overrides the free-form text to describe the transaction.
+In ledger2beancount, we offer several features to determine the payee.
+
+You can set `payee_split` and define a list of regular expressions which
+allow you to split ledger's payee field into payee and narration.  You
+have to use named capture groups `payee` and `narration`.  For example,
+
+    (?<narration>.*?)\s+\((?<payee>Tesco)\)
+
+allows you to split `Supermarket (Tesco)` into payee `Tesco` and
+narration `Supermaket`.  `payee_split` is a list of regular expressions
+and we stop once we have a match.
+
+Furthermore, you can use `payee_match` to match based on the ledger
+payee field and assign payees according to the match.  This variable
+is a hash consisting of a regular expressions and the corresponding
+payees.  For example, if your ledger contains a transaction like:
+
+    2018-03-18 * BART top-up
+
+you can use
+
+    BART top-up: BART
```

Comment:
before reading the code, it took me a while to understand how this example was supposed to work.
I think it's because for neither `payee_split` nor `payee_match` we give a complete example. I suggest changing the examples so that you also see the key names in action, rather than only the values.